packages/nx-adsp/src/generators/angular-app/angular-app.ts

```diff
@@ -0,0 +1,166 @@
+    templateOptions
+  );
+  const addProxyConf = options.nginxProxies.length > 0;
+  if (addProxyConf) {
+    // Add a webpack dev server proxy configuration ...
+    // TODO: Is this cleaner to add via template?
+    const devProxyConf = options.nginxProxies.reduce(
+      (proxyConf, nginxProxy) => {
+        const upstreamUrl = new URL(nginxProxy.proxyPass);
+
+        const proxy = {
+          target: `${upstreamUrl.protocol}//localhost${upstreamUrl.port ? ':' + upstreamUrl.port : ''}`,
+          secure: upstreamUrl.protocol === 'https:',
+          changeOrigin: false,
+          pathRewrite: {}
+        }
+
+        // If there is a path on the upstream url, then add a rewrite.
+        if (upstreamUrl.pathname.length > 1) {
+          proxy.pathRewrite = {
+            [`^${nginxProxy.location}`]: upstreamUrl.pathname
+          }
+        }
+
+        return {
+          ...proxyConf,
+          [nginxProxy.location]: proxy
+        }
+      },
+      {}
+    );
+    writeJson(
+      host,
+      `${options.projectRoot}/proxy.conf.json`,
+      devProxyConf
+    );
+  }
+  return addProxyConf;
+}
+
+function removeFiles(host: Tree, options: NormalizedSchema) {
+  host.delete(`${options.projectRoot}/src/app/logo.svg`);
+  host.delete(`${options.projectRoot}/src/app/star.svg`);
+}
+
+export default async function (host: Tree, options: AngularAppGeneratorSchema) {
+  const normalizedOptions = normalizeOptions(host, options);
+  const initAngular = wrapAngularDevkitSchematic('@nrwl/angular', 'application');
+
+  await initAngular(host, options);
+
+  addDependenciesToPackageJson(
+    host,
+    {
+    },
+    {
+      '@abgov/core-css': '^0.7.56',
+      '@abgov/angular-components': '1.0.1',
+      'html-webpack-plugin': '~4.5.2',
+      'oidc-client': '~1.11.5',
+    }
+  )
+
+  const addedProxy = addFiles(host, normalizedOptions);
+  removeFiles(host, normalizedOptions);
+
+  const layout = getWorkspaceLayout(host);
+
+  const config = readProjectConfiguration(host, options.name);
+
+  config.targets.build.options = {
+    ...config.targets.build.options,
+    assets: [
+      ...config.targets.build.options.assets,
+      {
+        "glob": "nginx.conf",
+        "input": `${layout.appsDir}/${options.name}`,
+        "output": "./"
+      }
+    ],
+  }
+
+  if (addedProxy) {
+    // Add the webpack dev server proxy if there is proxy configuration.
+    config.targets.serve.options = {
+      ...config.targets.serve.options,
+      proxyConfig: `${normalizedOptions.projectRoot}/proxy.conf.json`
+    }
+  }
+
+  updateProjectConfiguration(host, options.name, config);
+
+  await formatFiles(host);
+
+  if (hasDependency(host, '@abgov/nx-oc')) {
+    const { deploymentGenerator } = await import(`${'@abgov/nx-oc'}`);
+    await deploymentGenerator(
+      host,
+      {
+        ...options,
+        project: normalizedOptions.projectName
+      }
+    );
+  }
+
+  // return () => {
+  //   installPackagesTask(host);
+  // }
```

Comment:
The diff if I did add this

$ diff package.json ../stuff29/package.json
2c2
<   "name": "stuff27",
---
>   "name": "stuff29",
26c26,27
<     "help": "nx help"
---
>     "help": "nx help",
>     "postinstall": "ngcc --properties es2015 browser module main"
30,36c31,41
<     "core-js": "^3.6.5",
<     "document-register-element": "1.13.1",
<     "react": "17.0.2",
<     "react-dom": "17.0.2",
<     "regenerator-runtime": "0.13.7",
<     "styled-jsx": "3.4.4",
<     "tslib": "^2.0.0"
---
>     "@angular/animations": "^11.2.0",
>     "@angular/common": "^11.2.0",
>     "@angular/compiler": "^11.2.0",
>     "@angular/core": "^11.2.0",
>     "@angular/forms": "^11.2.0",
>     "@angular/platform-browser": "^11.2.0",
>     "@angular/platform-browser-dynamic": "^11.2.0",
>     "@angular/router": "^11.2.0",
>     "rxjs": "~6.6.3",
>     "tslib": "^2.0.0",
>     "zone.js": "^0.10.2"
39,42c44,53
<     "@babel/core": "7.12.13",
<     "@babel/preset-env": "7.12.13",
<     "@babel/preset-react": "7.12.13",
<     "@babel/preset-typescript": "7.12.13",
---
>     "@abgov/angular-components": "1.0.1",
>     "@abgov/core-css": "^0.7.56",
>     "@abgov/nx-adsp": "file:../stuff2/nx-tools/dist/packages/nx-adsp",
>     "@angular-devkit/build-angular": "~0.1102.0",
>     "@angular-eslint/eslint-plugin": "~4.2.0",
>     "@angular-eslint/eslint-plugin-template": "~4.2.0",
>     "@angular-eslint/template-parser": "~4.2.0",
>     "@angular/compiler-cli": "^11.2.0",
>     "@angular/language-service": "^11.2.0",
>     "@nrwl/angular": "^12.1.0",
48d58
<     "@nrwl/react": "12.1.0",
50d59
<     "@nrwl/web": "12.1.0",
52d60
<     "@testing-library/react": "11.2.6",
55,57d62
<     "@types/react": "17.0.3",
<     "@types/react-dom": "17.0.3",
<     "@types/styled-jsx": "2.2.8",
60d64
<     "babel-jest": "26.2.2",
66,69c70
<     "eslint-plugin-import": "2.22.1",
<     "eslint-plugin-jsx-a11y": "6.4.1",
<     "eslint-plugin-react": "7.23.1",
<     "eslint-plugin-react-hooks": "4.2.0",
---
>     "html-webpack-plugin": "~4.5.2",
70a72,73
>     "jest-preset-angular": "8.4.0",
>     "oidc-client": "~1.11.5",